### PR TITLE
TF-TRT Dynamic shape mode test for Softmax op converter

### PR DESCRIPTION
**Pull request by tfeher**
This PR adds explicit batch and dynamic shape mode test for the softmax op converter. 

Tagging @bixia1 for review and @DEKHTIARJonathan for visibility

Tracker: #45481

### CODE REVIEW

File: tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc [hunk 1]
```diff
@@ -3480,6 +3480,37 @@ TEST_P(OpConverter_FP32_Test, ConvertExpandDims) {
   }
 }
 
+TEST_P(OpConverter_FP32_FP16_Test, ConvertSoftmax) {
+  // Get the NodeDef for SoftMax.
+  Scope s = Scope::NewRootScope();
+  auto input = ops::Placeholder(s.WithOpName("logits"), tf_type_);
+  auto transpose = ops::Softmax(s.WithOpName("my_softmax"), input);
+  const NodeDef& node_def = transpose.operation.node()->def();
```
Comment (bixia1) [inline tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc:3488]: Nit: why we want to call this "transpose"?

File: tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc [hunk 2]
```diff
@@ -3480,6 +3480,37 @@ TEST_P(OpConverter_FP32_Test, ConvertExpandDims) {
...
+    Status runtime_status;
+  };
+  std::vector<TestParams> test_params = {
+      TestParams{{2, 3},
+                 {0.09003057, 0.24472848, 0.66524094, 0.09003057, 0.24472848,
+                  0.66524094}},
+      TestParams{{6, 1}, {1, 1, 1, 1, 1, 1}},  // works with std input
+      TestParams{{1, 6},  // this works with arange(1,7) input
+                 {0.00426978, 0.01160646, 0.03154963, 0.08576079, 0.23312202,
+                  0.6336913}},
```
Comment (bixia1) [inline tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc:3503]: Sorry I don't see how are the values for "conversion_status" and "runtime_status" set?

Also, do we want to test different values for [the "axis" of the op ](https://github.com/tensorflow/tensorflow/blob/dcce8453cf036c668a41d8435a3c087d6012faa5/tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc#L5632) ?

**Comment by gbaned** [comment]
@tfeher Can you please check @bixia1's comments and keep us posted ? Thanks!

**Comment by tensorflowbutler** [comment]
It has been 15 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

**Comment by tfeher** [comment]
The PR is still valid, will update this in a few days.

File: tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc (hunk 1 above)
Comment (tfeher) [reply tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc]: Corrected. It was copy-paste error.

File: tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc (hunk 2 above)
Comment (tfeher) [reply tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc:3651]: conversion_status and runtime_status are value initialized according to c++ aggregate initialization rules. But they are not used, so I have removed them.

I do not see any axis parameter for the [TF softmax op](https://www.tensorflow.org/api_docs/cc/class/tensorflow/ops/softmax), only the TRT layer has that param. The TF op assumes 2D input, first dim is the batch dim, second is the softmax dim.

**Review by tfeher — COMMENTED** [review]
Thanks @bixia1 for the review, I have addressed the issues.

**Review by bixia1 — APPROVED** [review]
